Jellyfin.Api/Controllers/Images/RemoteImageController.cs

```diff
@@ -0,0 +1,290 @@
+        /// <param name="startIndex">Optional. The record index to start at. All items with a lower index will be dropped from the results.</param>
+        /// <param name="limit">Optional. The maximum number of records to return.</param>
+        /// <param name="providerName">Optional. The image provider to use.</param>
+        /// <param name="includeAllLanguages">Optinal. Include all languages.</param>
+        /// <returns>Remote Image Result.</returns>
+        [HttpGet("{Id}/RemoteImages")]
+        [ProducesResponseType(typeof(RemoteImageResult), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetRemoteImages(
+            [FromRoute] string id,
+            [FromQuery] ImageType? type,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit,
+            [FromQuery] string providerName,
+            [FromQuery] bool includeAllLanguages)
+        {
+            try
+            {
+                var item = _libraryManager.GetItemById(id);
+                if (item == null)
+                {
+                    return NotFound();
+                }
+
+                var images = await _providerManager.GetAvailableRemoteImages(
+                        item,
+                        new RemoteImageQuery
+                        {
+                            ProviderName = providerName,
+                            IncludeAllLanguages = includeAllLanguages,
+                            IncludeDisabledProviders = true,
+                            ImageType = type
+                        }, CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                var imageArray = images.ToArray();
+                var allProviders = _providerManager.GetRemoteImageProviderInfo(item);
+                if (type.HasValue)
+                {
+                    allProviders = allProviders.Where(o => o.SupportedImages.Contains(type.Value));
+                }
+
+                var result = new RemoteImageResult
+                {
+                    TotalRecordCount = imageArray.Length,
+                    Providers = allProviders.Select(o => o.Name)
+                        .Distinct(StringComparer.OrdinalIgnoreCase)
+                        .ToArray()
+                };
+
+                if (startIndex.HasValue)
+                {
+                    imageArray = imageArray.Skip(startIndex.Value).ToArray();
+                }
+
+                if (limit.HasValue)
+                {
+                    imageArray = imageArray.Take(limit.Value).ToArray();
+                }
+
+                result.Images = imageArray;
+                return Ok(result);
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
```

Comment:
This would be better in a separate middleware. Otherwise most of the controllers will have the same try-catch boilerplate everywhere.